### ISSUE TITLE
Question about the pin slack

### ISSUE THREAD

**Issue by cai-zhi-jie** [liangrj2014/ISPD25_contest#13]
I wonder what the definition of pin slack is?
Is minimum slack of all timing path passing the pin?

If so, I assume that the slack of source pin should be the minimum among all pins in a net.

But I found some exceptions in ariane.
Some sink pin's slack is slightly less than the source pin (negtive or positive slack) with the difference in the last decimal place.
![fc02dbbf2d76589dc4569ad579c0770](https://github.com/user-attachments/assets/30f22c3a-9c41-41ab-a505-abb25c007394)
![image](https://github.com/user-attachments/assets/0fa030f1-fd3d-4c1a-8af7-9409ff835955)

Is this phenomenon normal？

**Comment by liangrj2014** [comment]
Yes, the definition of pin slack is "the minimum slack of all timing path passing the pin". I'm unsure of the cause of the mismatch, but I hope this minor discrepancy doesn't impact the algorithm or the scores.
